bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/update/InstallToolsHandler.java

```diff
@@ -215,18 +214,19 @@ protected void handleToolsInstallPython()
 	
 	protected void handleWebSocketClientInstall()
 	{
-		IPath pipPath = new org.eclipse.core.runtime.Path(pythonExecutablenPath); //$NON-NLS-1$
-		String pipPathLastSegment = pipPath.lastSegment().replace("python", "pip"); //$NON-NLS-1$ //$NON-NLS-2$
-		pipPath = pipPath.removeLastSegments(1).append(pipPathLastSegment); 
-		if (!pipPath.toFile().exists()) 
+		 
+		// pip install websocket-client
+		List<String> arguments = new ArrayList<String>();
+		final String pythonEnvPath = IDFUtil.getIDFPythonEnvPath();
+		if (pythonEnvPath != null && new File(pythonEnvPath).exists())
 		{
-			console.println(String.format("%s executable not found. Unable to run `%s install websocket-client`", pipPathLastSegment, pipPathLastSegment)); //$NON-NLS-1$
+			console.println(String.format("%s executable not found. Unable to run `%s -m pip install websocket-client`", //$NON-NLS-1$
+					IDFConstants.PYTHON_CMD, IDFConstants.PYTHON_CMD));
```

Comment:
I think this is the case for **not** where the python file does not exist.

Reply:
Hi @kolipakakondal. Thanks for noticing this. Fixed in the latest commit